awssql/plugins/bg/connect_routing.go

```diff
@@ -0,0 +1,241 @@
+	}
+	message := error_util.GetMessage("BlueGreenDeployment.switchoverCompleteContinueWithConnect", time.Since(holdStartTime))
+	slog.Debug(message)
+
+	return nil, error_util.NewGenericAwsWrapperError(message)
+}
+
+func NewSuspendConnectRouting(hostAndPort string, role driver_infrastructure.BlueGreenRole, bgId string) *SuspendConnectRouting {
+	return &SuspendConnectRouting{bgId, NewBaseRouting(hostAndPort, role)}
+}
+
+type SuspendUntilCorrespondingHostFoundConnectRouting struct {
+	bgId string
+	BaseRouting
+}
+
+func (r *SuspendUntilCorrespondingHostFoundConnectRouting) Apply(_ driver_infrastructure.ConnectionPlugin, hostInfo *host_info_util.HostInfo, props map[string]string,
+	_ bool, pluginService driver_infrastructure.PluginService) (driver.Conn, error) {
+	slog.Debug(error_util.GetMessage("BlueGreenDeployment.waitConnectUntilCorrespondingHostFound", hostInfo.Host))
+	parentCtx := pluginService.GetTelemetryContext()
+	telemetryFactory := pluginService.GetTelemetryFactory()
+	telemetryCtx, ctx := telemetryFactory.OpenTelemetryContext(TELEMETRY_SWITCHOVER, telemetry.NESTED, parentCtx)
+
+	pluginService.SetTelemetryContext(ctx)
+	defer func() {
+		telemetryCtx.CloseContext()
+		pluginService.SetTelemetryContext(parentCtx)
+	}()
+
+	bgStatus, ok := pluginService.GetStatus(r.bgId)
+	var correspondingPair utils.Pair[*host_info_util.HostInfo, *host_info_util.HostInfo]
+	if ok && !bgStatus.IsZero() {
+		correspondingPair = bgStatus.GetCorrespondingHosts()[hostInfo.Host]
+	}
+
+	timeoutMs := property_util.GetVerifiedWrapperPropertyValue[int](props, property_util.BG_CONNECT_TIMEOUT_MS)
+	holdStartTime := time.Now()
+	endTime := holdStartTime.Add(time.Millisecond * time.Duration(timeoutMs))
+
+	for time.Now().Before(endTime) && ok && !bgStatus.IsZero() && bgStatus.GetCurrentPhase() != driver_infrastructure.COMPLETED &&
+		correspondingPair.GetRight().IsNil() {
+		r.Delay(SLEEP_TIME_DURATION, bgStatus, pluginService, r.bgId)
+		bgStatus, ok = pluginService.GetStatus(r.bgId)
+		if ok && !bgStatus.IsZero() {
+			correspondingPair = bgStatus.GetCorrespondingHosts()[hostInfo.Host]
+		}
+	}
+
+	if bgStatus.IsZero() || bgStatus.GetCurrentPhase() == driver_infrastructure.COMPLETED {
+		message := error_util.GetMessage("BlueGreenDeployment.completedContinueWithConnect", time.Since(holdStartTime))
+		slog.Debug(message)
+		return nil, error_util.NewGenericAwsWrapperError(message)
+	} else if time.Now().After(endTime) {
+		return nil, error_util.NewGenericAwsWrapperError(error_util.GetMessage("BlueGreenDeployment.correspondingHostNotFoundTryConnectLater", hostInfo.Host, timeoutMs))
+	}
+
+	message := error_util.GetMessage("BlueGreenDeployment.correspondingHostFoundContinueWithConnect", hostInfo.Host, time.Since(holdStartTime))
+	slog.Debug(message)
+	// review if we shpould be returning an err here
+
+	return nil, error_util.NewGenericAwsWrapperError(message)
```

Comment:
Going to restate your comment lol, but will this function always return an error?